server/portal/apps/workspace/api/views.py

```diff
@@ -44,16 +46,29 @@ def _get_user_app_license(license_type, user):
     return lic
 
 
+def _get_exec_systems(user, systems):
+    """List of all enabled execution systems available for the user."""
+    tapis = user.tapis_oauth.client
+    system_id_search = ','.join(systems)
+    search_string = f"(id.in.{system_id_search})~(canExec.eq.true)~(enabled.eq.true)"
+    return tapis.systems.getSystems(listType="ALL", select="allAttributes", search=search_string)
+
+
 def _get_app(app_id, app_version, user):
     tapis = user.tapis_oauth.client
     if app_version:
         app_def = tapis.apps.getApp(appId=app_id, appVersion=app_version)
     else:
         app_def = tapis.apps.getAppLatestVersion(appId=app_id)
+
     data = {'definition': app_def}
 
-    # GET EXECUTION SYSTEM INFO TO PROCESS SPECIFIC SYSTEM DATA E.G. QUEUE INFORMATION
-    data['exec_sys'] = tapis.systems.getSystem(systemId=app_def.jobAttributes.execSystemId)
+    exec_systems = getattr(app_def.notes, 'dynamicExecSystems', TapisResult()).__dict__.keys()
```

Comment:
I think we landed on keeping it optional.

Reply:
Rather, I mean if `dynamicExecSystem` is `true`, and `dynamicExecSystems` is undefined, should that mean that we do no filtering on the systems listed in the dropdown?

Reply:
Oh, I see. We wanted to see if one attribute could provide everything.

So, if we choose dynamicExecSystems, then:

* dynamicExecSystems: ["All"] - would do no filtering, would pick all systems available to the user.
* "dynamicExecSystems":  map of systems to system label will limit to these systems only.

 I can make that change, it would be useful for some apps and not require provisioning of app if a new system is added.
How would we want to track labels in this case. An example is ls6 - exec systems are defined as ls6. There is no end user friendly label is system definition. The only other way is to maintain a global list of system names to label. 
This would fit names of well known system names and their labels. Should use notes field in [system definition](https://github.com/tapis-project/tapipy/blob/main/tapipy/resources/openapi_v3-systems.yml#L2182) for label?

Reply:
@rstijerina  - The fix for this is already in, please let me know if this addressed this issue.
There will be only 1 property `dynamicExecSystems`, it will be handled:
* if the property does not exist, then there the exec system is static and used as defined in app json.
* If `dynamicExecSystems` is `All`, then all the available to the user are provided in the UI.
* If `dynamicExecSystems` is a list of exec systems, then the list is used if the user has access to the system.